### PR TITLE
[FT] CF-171 Validate deleted images are restored only once

### PR DESCRIPTION
**Pull request by ntymtsiv**
Scenario:
    1. Get deleted and broken image's ids from src
    2. Get all images from dst
    3. Verify each deleted and broken image has been restored once

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: devlab/tests/testcases/test_resource_migration.py [hunk 1]
```diff
@@ -334,6 +334,39 @@ def test_migrate_glance_images(self):
...
+        # getting 'broken' images (which exist in the glance, but deleted in
+        # storage)
+        all_images = self.migration_utils.get_all_images_from_config()
+        broken_images = [i['name'] for i in all_images if i.get('broken')]
+        src_images = self.src_cloud.glanceclient.images.list()
+        to_restore_img_ids.extend([image.id for image in src_images
+                                  if image.name in broken_images])
+
+        dst_images = [x for x in self.dst_cloud.glanceclient.images.list()]
+        restored_dst_images = []
```
Comment (roman-verchikov) [inline devlab/tests/testcases/test_resource_migration.py]: maybe `defaultdict` instead?

File: devlab/tests/testcases/test_resource_migration.py [hunk 2]
```diff
@@ -334,6 +334,39 @@ def test_migrate_glance_images(self):
...
+        for deleted_img_id in set(to_restore_img_ids):
+            for dst_image in dst_images:
+                if dst_image.name and deleted_img_id in dst_image.name:
+                    restored_dst_images.append(dst_image.name)
+        duplicated = [i for i in restored_dst_images
+                      if restored_dst_images.count(i) > 1]
+        if duplicated:
+            msg = 'Following images were restored more than 1 time: "%s"'
```
Comment (roman-verchikov) [inline devlab/tests/testcases/test_resource_migration.py]: Would be more readable with number of actual restores, e.g. "Image X restored N times"

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
Please rebase

**Comment by roman-verchikov** [comment]
Please add test scenario to pull request description and commit message

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
Looks like you need to modify `generate_load.py` too, because it fails to create VM with `deleted_image` image:

```
2016-03-15 16:51:39,956 - INFO - Creating vms
Traceback (most recent call last):
  File "devlab/generate_load.py", line 1079, in <module>
    preqs.run_preparation_scenario()
  File "devlab/generate_load.py", line 1010, in run_preparation_scenario
    self.create_vms()
  File "devlab/generate_load.py", line 415, in create_vms
    vms.extend(create_vms(keypairs_vms[keypair]))
  File "devlab/generate_load.py", line 378, in create_vms
    **self._get_parameters_for_vm_creating(vm))
  File "devlab/generate_load.py", line 327, in _get_parameters_for_vm_creating
    image_id = self.get_image_id(vm['image']) if vm.get('image') else ''
  File "/home/vagrant/CloudFerry/devlab/tests/base.py", line 123, in get_image_id
    raise NotFound('Image with name "%s" was not found' % image_name)
devlab.tests.test_exceptions.NotFound: Image with name "deleted_image" was not found
```

**Comment by ntymtsiv** [comment]
It was wrong rebased

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: devlab/tests/testcases/test_resource_migration.py [hunk 3]
```diff
@@ -339,6 +340,46 @@ def test_migrate_glance_images(self):
...
+    def test_migrate_deleted_glance_images_only_once(self):
+        """Validate deleted and broken images were migrated to dst only once.
+
+        Scenario:
+            1. Get deleted and broken image's ids from src
+            2. Get all images from dst
+            3. Verify each deleted and broken image has been restored once
+        """
+        src_vms = self.src_cloud.novaclient.servers.list(
+            search_opts={'all_tenants': 1})
```
Comment (SVilgelm) [inline devlab/tests/testcases/test_resource_migration.py]: Use `True`

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
+1

**Comment by ntymtsiv** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
can be merged

**Pull request merged**